### PR TITLE
Add update-codec-server

### PR DESCRIPTION
**Pull request by shivam-ajmera**
## What was changed
Added a tcld command.

## Why?
To update codec server config via tcld

## Checklist

1. Closes 

2. How was this tested:
Unit test

3. Any docs updates needed?

### CODE REVIEW

File: app/namespace_test.go [hunk 1]
```diff
@@ -1063,6 +1064,117 @@ func (s *NamespaceTestSuite) TestClearCertificateFilters() {
...
+			expectGet: func(g *namespaceservice.GetNamespaceResponse) {},
+			expectUpdate: func(r *namespaceservice.UpdateNamespaceRequest) {
+				r.Spec.CodecSpec = &namespace.CodecServerPropertySpec{
+					Endpoint:           "fakehost:9999",
+					PassAccessToken:    true,
+					IncludeCredentials: true,
+				}
+			},
+		},
+		// {
```
Comment (jumbogo) [inline app/namespace_test.go:1115]: If this is not used any more, just remove all these commented code.

File: app/namespace.go [hunk 2]
```diff
@@ -829,6 +829,63 @@ func NewNamespaceCommand(getNamespaceClientFn GetNamespaceClientFn) (CommandOut,
...
+				NamespaceFlag,
+				&cli.StringFlag{
+					Name:     "endpoint",
+					Usage:    "The codec server endpoint to decode payloads for all users interacting with this Namespace.",
+					Aliases:  []string{"e"},
+					Required: true,
+				},
+				&cli.BoolFlag{
+					Name:    "pass-access-token",
+					Usage:   "Pass the user access token with your endpoint.",
```
Comment (mastermanu) [inline app/namespace.go:846]: "to your endpoint" ?

File: app/namespace.go [hunk 3]
```diff
@@ -829,6 +829,63 @@ func NewNamespaceCommand(getNamespaceClientFn GetNamespaceClientFn) (CommandOut,
...
 		},
+		{
+			Name:    "update-codec-server",
+			Usage:   "Update codec server config used to decode your encoded payloads through remote endpoint.",
+			Aliases: []string{"ucs"},
+			Flags: []cli.Flag{
+				NamespaceFlag,
+				&cli.StringFlag{
+					Name:     "endpoint",
+					Usage:    "The codec server endpoint to decode payloads for all users interacting with this Namespace.",
```
Comment (mastermanu) [inline app/namespace.go:840]: [Nit] I think usage fields typically don't end with punctuation, at least with respect to the rest of the file, right?

File: app/namespace_test.go (hunk 1 above)
Comment (shivam-ajmera) [reply app/namespace_test.go:1115]: Oops did not catch that, thank you!

File: app/namespace.go (hunk 3 above)
Comment (shivam-ajmera) [reply app/namespace.go]: You're right, changed it to match the rest

File: app/namespace.go [hunk 4]
```diff
@@ -829,6 +829,63 @@ func NewNamespaceCommand(getNamespaceClientFn GetNamespaceClientFn) (CommandOut,
...
+					PassAccessToken:    ctx.Bool("pass-access-token"),
+					IncludeCredentials: ctx.Bool("include-credentials"),
+				}
+
+				difference, err := compareCodecSpec(n.Spec.CodecSpec, replacement)
+				if err != nil {
+					return err
+				}
+
+				fmt.Println("this import will result in the following changes to the codec server config:")
```
Comment (jumbogo) [inline app/namespace.go:872]: [nit] "this import" seems confusing, change to "this update"?

**Review by jumbogo — APPROVED** [review]
Looks good.  Have some "nits".  Also maybe ping product to see if the command line prompt language is good.

File: app/namespace.go [hunk 5]
```diff
@@ -829,6 +829,63 @@ func NewNamespaceCommand(getNamespaceClientFn GetNamespaceClientFn) (CommandOut,
...
 		},
+		{
+			Name:    "update-codec-server",
+			Usage:   "Update codec server config used to decode encoded payloads through remote endpoint",
+			Aliases: []string{"ucs"},
+			Flags: []cli.Flag{
+				NamespaceFlag,
+				&cli.StringFlag{
+					Name:     "endpoint",
+					Usage:    "The codec server endpoint to decode payloads for all users interacting with this Namespace",
```
Comment (jlacefie) [inline app/namespace.go:840]: IIUC this must be in https form.  Will you please ensure that the description mentions https is required?

File: app/namespace.go [hunk 6]
```diff
@@ -829,6 +829,63 @@ func NewNamespaceCommand(getNamespaceClientFn GetNamespaceClientFn) (CommandOut,
...
+				},
+			},
+			Action: func(ctx *cli.Context) error {
+				n, err := c.getNamespace(ctx.String(NamespaceFlagName))
+				if err != nil {
+					return err
+				}
+
+				replacement := &namespace.CodecServerPropertySpec{
+					Endpoint:           ctx.String("endpoint"),
```
Comment (jlacefie) [inline app/namespace.go:862]: can we add a validation check to ensure https URL please?

**Review by jlacefie — COMMENTED** [review]
Added 2 notes about adding https to the update-codec-server --help and validation logic

**Pull request merged**